pkg/executor/chained_task_handle.go

```diff
@@ -0,0 +1,112 @@
+package executor
+
+import (
+	"sync"
+	"time"
+)
+
+// ChainedTaskHandle is an links Launchers in a way that
+// one will be launched after another.
+type ChainedTaskHandle struct {
+	TaskHandle
+
+	chainedLauncher Launcher
+
+	encounteredError error
+
+	chainFinished chan struct{}
+	stopChain     chan struct{}
+	stopOnce      sync.Once
+}
+
+// NewChainedTaskHandle returns TaskHandle that executes current handle, and will
+// launch Launcher when handle will finish it's execution.
+func NewChainedTaskHandle(handle TaskHandle, launcher Launcher) TaskHandle {
```

Comment:
@skonefal consider using variadic instead of single `Laucher`. It would be much more flexible.